File: js/primeNumber.js
function isPrimeNumber(n) {
  if (n < 2) return false;
  if (n == 2) return true;

  for (let i = 2; i < n - 1; i++) {
    if (n % i === 0) return false;
  }

  return true;
}

// BIG O = O(n)

console.log(isPrimeNumber(2));
console.log(isPrimeNumber(7));
console.log(isPrimeNumber(3));
console.log(isPrimeNumber(15));
console.log(isPrimeNumber(6));
console.log("\n=================\n");

function isPrime(n) {
  if (n < 2) return false;
  for (let i = 2; i <= Math.sqrt(n); i++) {
    if (n % i === 0) {
      return false;
    }
  }
  return true;
}

// Big O = O(sqrt(n))

console.log(isPrime(2));
console.log(isPrime(7));
console.log(isPrime(3));
console.log(isPrime(15));
console.log(isPrime(49));

File: python/primeNumber.py
import math


def isPrimeNumber(n):
    if n < 2:
        return False
    if n == 2:
        return True

    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False

    return True


print(isPrimeNumber(5))
print(isPrimeNumber(15))
print(isPrimeNumber(3))
print(isPrimeNumber(21))
